common/script/index.coffee

```diff
@@ -646,6 +646,20 @@ api.wrap = (user, main=true) ->
         cb? null, user.preferences.webhooks
 
       # ------
+      # Push Notifications
+      # ------
+      addPushDevice: (req, cb) ->
+        user.pushDevices = [] unless user.pushDevices
+        pd = user.pushDevices
+        item = {regId:req.body.regId, type:req.body.type};
+        i = _.findIndex pd, {regId: item.regId}
+
+        pd.push(item) unless i != -1
+
+        user.markModified? 'pushDevices'
```

Comment:
the only thing here is that since `addPushDevice` is server-side only I'm not sure if it would be better to put it in a controller instead of index.coffee